src/summarizeGitHubCommits.ts

```diff
@@ -0,0 +1,114 @@
+import { TriggerClient, cronTrigger } from "@trigger.dev/sdk";
+import { Github } from "@trigger.dev/github";
+import { OpenAI } from "@trigger.dev/openai";
+import { Slack } from "@trigger.dev/slack";
+
+const client = new TriggerClient({ id: "jobs-showcase" });
+
+const github = new Github({
+  id: "github",
+  token: process.env.GITHUB_TOKEN!,
+});
+
+const openai = new OpenAI({
+  id: "openai",
+  apiKey: process.env.OPENAI_API_KEY!,
+});
+
+const slack = new Slack({ id: "slack" });
+
+// This Job runs at 7am every day using a cron schedule expression.
+client.defineJob({
+  id: "summarize-yesterday-commits-on-slack",
+  name: "Summarize yesterday's commits on Slack",
+  version: "1.0.0",
+  trigger: cronTrigger({
+    // Note that the time is in UTC.
+    // Converted to cron schedule expression using https://crontab.guru/
+    cron: "0 7 * * *",
+  }),
+  integrations: {
+    github,
+    openai,
+    slack,
+  },
+  run: async (payload, io, ctx) => {
+    // 1. Calculate the 'since' and 'until' timestamps for yesterday.
+    const timestamps = await io.runTask("get-timestamps", async () => {
```

Comment:
You don't need this, as the payload has a `ts` and `lastTimestamp` value that you can use for this exact purpose:

![CleanShot 2023-10-04 at 09 18 08](https://github.com/triggerdotdev/jobs-showcase/assets/534/f4071af4-16d9-4931-9991-82fdff6a460d)